lib/simple-seeder.js

```diff
@@ -0,0 +1,221 @@
+      await this._remove(key)
+      return this._add(key, opts)
+    }
+
+    if (!!info.seeders !== opts.seeders) {
+      if (opts.seeders) {
+        if (info.seeders) throw new Error('Seeders was already enabled') // Should not happen
+        const publicKey = HypercoreId.decode(info.key)
+        info.seeders = await this._createSeeders(publicKey)
+      } else {
+        await info.seeders.destroy()
+        info.seeders = null
+      }
+    }
+
+    info.description = opts.description || null
+
+    return info
+  }
+
+  async remove (key) {
+    const release = await this.lock()
+    try {
+      return await this._remove(key)
+    } finally {
+      release()
+    }
+  }
+
+  async _remove (key) {
+    const info = this.get(key)
+
+    if (info.seeders) await info.seeders.destroy()
+    if (info.discovery) await info.discovery.destroy()
+    await info.instance.close()
+
+    this.resources.delete(info.key)
+  }
+
+  has (key) {
+    const id = HypercoreId.normalize(key)
+    return this.resources.has(id)
+  }
+
+  get (key) {
+    const id = HypercoreId.normalize(key)
+    const info = this.resources.get(id)
+    if (!info) throw new Error('Resource does not exists')
+    return info
+  }
+
+  filter (onfilter) {
+    const arr = []
+    for (const info of this.values()) {
+      if (onfilter(info)) arr.push(info)
+    }
+    return arr
+  }
+
+  values () {
+    return this.resources.values()
+  }
+
+  async destroy () {
+    if (this.destroying) return
+    this.destroying = true
+
+    const closing = []
+    for (const info of this.values()) {
+      closing.push(info.destroy())
+    }
+    await Promise.all(closing)
+  }
+
+  async _createCore (publicKey, info, opts = {}) {
+    const core = this.store.get(publicKey)
+    await core.ready()
+
+    this._suitCore(core, info, opts)
+
+    return core
+  }
+
+  async _createBee (publicKey, info, opts = {}) {
+    const core = this.store.get(publicKey)
+    const bee = new Hyperbee(core, { keyEncoding: opts.keyEncoding, valueEncoding: opts.valueEncoding })
+    await bee.ready()
+
+    this._suitCore(bee.core, info, opts)
+
+    return bee
+  }
+
+  async _createDrive (publicKey, info, opts = {}) {
+    const drive = new Hyperdrive(this.store, publicKey)
+    await drive.ready()
+
+    this._suitCore(drive.core, info, opts)
+
+    const onblobs = (blobs) => this._suitCore(blobs.core, info, opts)
+    if (drive.blobs) onblobs(drive.blobs)
+    else drive.once('blobs', onblobs)
+
+    if (opts.seeders) {
+      info.seeders = await this._createSeeders(publicKey)
+    }
+
+    return drive
+  }
+
+  async _createSeeders (publicKey) {
+    const id = HypercoreId.encode(publicKey)
+    const keyPair = await this.store.createKeyPair('simple-seeder-swarm@' + id)
+
+    const sw = new Seeders(publicKey, { dht: this.swarm.dht, keyPair })
+    sw.join()
+    sw.on('connection', (socket) => this.store.replicate(socket))
+
+    return sw
+  }
+
+  _suitCore (core, info, opts) {
```

Comment:
suit core sounds strange, if the concept is that of following a core, I would call it `followCore`.
The logic for this is also hard to follow, I can't figure out why `followSpeed` and `onSpeed` helper methods were created. From what I understand, it could be more explicit by just defining 2 handlers here:

```
  core.on('download', (...) => {   
    info.blocks['down'](1) })
    info.network['down'](byteLength)
  }
  core.on('upload', (...) => {
    info.blocks['up'](1) })
    info.network['up'](byteLength)
  })
```

Reply:
That still creates new functions and repeats the code twice, so I tend to prefer a bind reusing the same code

But no need for the `followSpeed` helper, will move it to the internal method

Reply:
I kind of think that Hypercore should do the `speedometer` internally so it just gives you the speeds